python/tk_houdini/base_hooks/node_handler_base.py

```diff
@@ -0,0 +1,471 @@
+
+        :rtype: An :class:`sgtk.Template` instance.
+        """
+        return self._work_template
+
+    def get_publish_template(self, node):
+        """
+        Get the shotgun publish template for this node handler.
+
+        :param node: A :class:`hou.Node` instance.
+
+        :rtype: An :class:`sgtk.Template` instance.
+        """
+        return self._publish_template
+        
+    def _get_template(self, template_name):
+        """
+        Get a shotgun template from the given name.
+
+        :param str template_name: The name of the template to get.
+
+        :rtype: An :class:`sgtk.Template` instance.
+
+        :raises: :class:`sgtk.TankError` if no template name supplied
+            or the template doesn't exist.
+        """
+        template_name = self.extra_args.get(template_name)
+        if not template_name:
+            raise sgtk.TankError("No template name defined")
+        template = self.parent.get_template_by_name(template_name)
+        if not template:
+            raise sgtk.TankError("Can't find template")
+        return template
+
+    #############################################################################################
+    # Utilities
+    #############################################################################################
+
+    def _remove_sgtk_items_from_parm_group(self, parameter_group):
+        """
+        Remove all sgtk parameters from the node's parameter template group.
+
+        :param ParmGroup parameter_group: The parameter group containing sgtk parameters.
+        """
+        raise NotImplementedError("'_remove_sgtk_items_from_parm_group' needs to be implemented")
+
+    def remove_sgtk_parms(self, node):
+        """
+        Remove all sgtk parameters from the node's parameter template group.
+
+        :param node: A :class:`hou.Node` instance containing sgtk parameters.
+        """
+        use_sgtk = node.parm(self.USE_SGTK)
+        if not use_sgtk:
+            return False
+        use_sgtk.set(False)
+        self._enable_sgtk(node, False)
+        parameter_group = self._get_parameter_group(node)
+        self._remove_sgtk_items_from_parm_group(parameter_group)
+        node.setParmTemplateGroup(parameter_group.build())
+        return True
+
+    def _restore_sgtk_parms(self, node):
+        """
+        Restore any removed sgtk parameters onto the given node.
+
+        :param node: A :class:`hou.Node` instance containing sgtk parameters.
+        """
+        raise NotImplementedError("'_restore_sgtk_parms' needs to be implemented")
+
+    def restore_sgtk_parms(self, node):
+        """
+        Restore any removed sgtk parameters onto the given node.
+
+        :param node: A :class:`hou.Node` instance containing sgtk parameters.
+        """
+        sgtk_identifier = node.parm(self.SGTK_IDENTIFIER)
+        use_sgtk = node.parm(self.USE_SGTK)
+        if use_sgtk or not sgtk_identifier:
+            return
+        self._restore_sgtk_parms(node)
```

Comment:
Simplify logic and remove extra exit point

```suggestion
        if sgtk_identifier and not use_sgtk:
            self._restore_sgtk_parms(node)
```